extensions/SpyroStudios/Arrays.js

```diff
@@ -0,0 +1,499 @@
+                defaultValue: 1
+              },
+              COLUMN: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 1
+              },
+              VALUE: {
+                type: ArgumentType.STRING,
+                defaultValue: "0"
+              }
+            }
+          },
+          {
+            opcode: "get2DArrayValue",
+            text: "value in 2D array [NAME] at row [ROW] column [COLUMN]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              },
+              ROW: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 1
+              },
+              COLUMN: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 1
+              }
+            }
+          },
+          {
+            opcode: "arrayExists2D",
+            text: "2D array [NAME] exists?",
+            blockType: BlockType.BOOLEAN,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          },
+          {
+            opcode: "arrayRowCount2D",
+            text: "row count of 2D array [NAME]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          },
+          {
+            opcode: "arrayColumnCount2D",
+            text: "column count of 2D array [NAME]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          },
+          {
+            opcode: "duplicateArray2D",
+            text: "duplicate 2D array [NAME] called [NEWNAME]",
+            blockType: BlockType.COMMAND,
+            arguments: {
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              },
+              NEWNAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "new_array"
+              }
+            }
+          },
+          {
+            opcode: "countOccurrencesOfItem2D",
+            text: "count occurrences of item [ITEM] in 2D array [NAME]",
+            blockType: BlockType.REPORTER,
+            arguments: {
+              ITEM: {
+                type: ArgumentType.STRING,
+                defaultValue: "item"
+              },
+              NAME: {
+                type: ArgumentType.STRING,
+                defaultValue: "array"
+              }
+            }
+          },
+          // 3D Array Blocks
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "3D Arrays",
+            hideFromPalette: false
```

Comment:
you don't need to tell the labels to not hide from the palette; they show by default anyway